src/test/java/com/open/boot/core/utils/CommonUtilTest.java

```diff
@@ -0,0 +1,142 @@
+                CommonUtil.getGenderByIdNumber("123456789012345678"));
+
+        Assert.assertEquals(new Integer(2),
+                CommonUtil.getGenderByIdNumber("000000000000000"));
+        Assert.assertEquals(new Integer(2),
+                CommonUtil.getGenderByIdNumber("000000000000000000"));
+    }
+
+    @Test
+    public void testGetBirthdayByIdNumber() {
+        Assert.assertEquals("", CommonUtil.getBirthdayByIdNumber("12345"));
+        Assert.assertEquals("1978-90-12",
+                CommonUtil.getBirthdayByIdNumber("123456789012345"));
+        Assert.assertEquals("7890-12-34",
+                CommonUtil.getBirthdayByIdNumber("123456789012345678"));
+    }
+
+    @Test
+    public void testGenRandomNum() {
+        Assert.assertEquals("", CommonUtil.genRandomNum(0));
+        Assert.assertEquals(5, CommonUtil.genRandomNum(5).length());
+        Assert.assertEquals(12, CommonUtil.genRandomNum(12).length());
+    }
+
+    @Test
+    public void testGetMapValue() {
+        HashMap<String, Object> hashMap = new HashMap<>();
+        hashMap.put("foo", "123");
+        Assert.assertEquals("123", CommonUtil.getMapValue(hashMap, "foo"));
+        Assert.assertEquals("", CommonUtil.getMapValue(hashMap, "baz"));
+    }
+
+    @Test
+    public void testGetSpecifiedDayBefore() {
+        Assert.assertEquals("2019-05-14",
+                CommonUtil.getSpecifiedDayBefore("2019-05-15"));
+    }
+
+    @Test
+    public void testGetSpecifiedDayAfter() {
+        Assert.assertEquals("2019-05-16",
+                CommonUtil.getSpecifiedDayAfter("2019-05-15"));
+    }
+
+    @PrepareForTest({CommonUtil.class})
+    @Test
+    public void testGetTimesmorning() throws Exception {
+        PowerMockito.mockStatic(Calendar.class);
+        SimpleDateFormat sdf = PowerMockito.mock(SimpleDateFormat.class);
+        PowerMockito.doReturn("19-05-15 00:00:00").when(sdf);
+        sdf.format(or(isA(Date.class), isNull(Date.class)));
+
+        PowerMockito.whenNew(SimpleDateFormat.class)
+                .withParameterTypes(String.class)
+                .withArguments(or(isA(String.class), isNull(String.class)))
+                .thenReturn(sdf);
+
+        Assert.assertEquals("19-05-15 00:00:00",
+                CommonUtil.getTimesmorning());
+    }
+
+    @PrepareForTest({CommonUtil.class})
+    @Test
+    public void testGetTimesnight() throws Exception {
+        PowerMockito.mockStatic(Calendar.class);
+        SimpleDateFormat sdf = PowerMockito.mock(SimpleDateFormat.class);
+        PowerMockito.doReturn("19-05-16 00:00:00").when(sdf);
+        sdf.format(or(isA(Date.class), isNull(Date.class)));
+
+        PowerMockito.whenNew(SimpleDateFormat.class)
+                .withParameterTypes(String.class)
+                .withArguments(or(isA(String.class), isNull(String.class)))
+                .thenReturn(sdf);
+
+        Assert.assertEquals("19-05-16 00:00:00",
+                CommonUtil.getTimesnight());
```

Comment:
In these last two tests, the functionality of the method isn't really tested. Instead of mocking `sdf`, you could mock `Calendar.getInstance` to always return a `Calendar` instance at a specific datetime (somewhere around the middle of a fixed date). That way the test would still verify the behaviour (e.g. setting the time to morning or evening).

Reply:
Thanks for the help on this :)